internal/app/networkd/pkg/networkd/networkd.go

```diff
@@ -285,11 +297,15 @@ func (n *Networkd) Hostname() (err error) {
 	}
 
 	if err = unix.Sethostname([]byte(hostname)); err != nil {
-		return err
+		// Workaround for container based setups
+		if err != syscall.EPERM {
+			return err
+		}
```

Comment:
I know we have done work to fine tune permissions, and this is unlikely to be an issue as long as we don't change permissions, but I feel strongly that a better route would be to skip setting the hostname and domain name if we are running in container mode. I can think of two ways to do this:

1. We add a flag to networkd that we can use to determine the runtime mode
2. We load the config and parse it to get the `runtime.Configurator`

Reply:
If we ignored EPERM for the entire networkd stack I'd agree it's too wide of a net. But we are ignoring it only for setting the domain name and hostname, two specific syscalls. I think this is reasonably tight. 

Having some notion of modes for networkd to run in feels like the wrong approach because it introduces another code branch which would introduce unnecessary complexity. Really the only thing we want from networkd in 'container' mode is setting up resolv.conf to prevent a loop error with coredns. So from that perspective I'd rather just drop networkd from the list of container services and bring back out patch for coredns to explicitly set the resolvers.

Reply:
Setting the hostname, and domain name, as trivial as they sound, are _extremely_ important. PKI is configured this way. Ignoring any kind of error on these should never happen.

I do like the idea of not running `networkd` in container mode. Especially with the work @smira is doing with firecracker. Remind me, how did we patch coredns? Feels like that was pre `osctl cluster create`. We would have to build that in somewhere in `osctl`. That makes it a little more ugly 🤔.